motionplan/constraint.go

```diff
@@ -493,3 +482,53 @@ func NewOctreeCollisionConstraint(octree *pointcloud.BasicOctree, threshold int,
 	}
 	return constraint
 }
+
+// NewBoundingRegionConstraint will determine if the given list of robot geometries are in collision with the
+// given list of bounding regions.
+func NewBoundingRegionConstraint(robotGeoms, boundingRegions []spatial.Geometry, collisionBufferMM float64) StateConstraint {
+	return func(state *ik.State) bool {
+		internalGeoms, b := obstacleAndBoudingRegionsConstraintHandler(robotGeoms, state, defaultBoundingRegionConstraintDesc)
+		if !b {
+			return b
+		}
+		cg, err := newCollisionGraph(internalGeoms, boundingRegions, nil, true, collisionBufferMM)
+		if err != nil {
+			return false
+		}
+		return len(cg.collisions(collisionBufferMM)) != 0
+	}
+}
+
+func obstacleAndBoudingRegionsConstraintHandler(
+	robotGeoms []spatial.Geometry, state *ik.State, constraintType string,
+) ([]spatial.Geometry, bool) {
+	var internalGeoms []spatial.Geometry
+	b := true
+	switch {
+	case state.Configuration != nil:
+		internal, err := state.Frame.Geometries(state.Configuration)
+		if err != nil {
+			b = false
+		}
+		internalGeoms = internal.Geometries()
+	case state.Position != nil:
```

Comment:
Hopefully, your work on 5391 should allow us to do away with this switch statement; could you add a TODO for that?